src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/StorePal.cs

```diff
@@ -109,7 +109,18 @@ public static IStorePal FromSystemStore(string storeName, StoreLocation storeLoc
                     if (ordinalIgnoreCase.Equals("Disallowed", storeName))
                         return AppleTrustStore.OpenStore(StoreName.Disallowed, storeLocation, openFlags);
 
-                    break;
+                    if (!IsValidStoreName(storeName))
```

Comment:
Is the "store creation is not supported on this platform" message really the best message for invalid store names?

Reply:
The message is actually "The platform does not have a definition for an X509 certificate store named '{0}' with a StoreLocation of '{1}', and does not support creating it.", which may not be perfect, but it is not wrong either.

Reply:
Well, and then it also carries a PlatformNotSupportedException within it.

The Linux version of not being a valid store name is at https://github.com/dotnet/corefx/blob/master/src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs#L313-L325

Which suggests that there's an IOException hiding here.

Reply:
Ah, I was sure I saw it in the Linux version, but I couldn't find now. I will follow that one then. Thanks!